035-ca-abstraction.md

```diff
@@ -134,8 +134,10 @@ Then the "CA certificate trust state machine":
 Now let's describe the processes.
 
 1. The issuing operator needs to figure out which EE certificates are missing (or in their renewal period, or otherwise need replacing, e.g. via annotation). This need for a certificate might exist independently of the end entity. For example, if we're going to scale up the Kafka cluster then we need a certificate for the new broker before the new broker exists. **The issuing operator puts the end entity in the `NEW` state.**
-2. (**See step 1 in the diagram.**) The issuing operator requests a new certificate. **The issuing operator puts the end entity in the `REQUESTED` state.**
-3. (**See step 2 in the diagram.**) The issuing operator cannot assume that issuance will happen quickly (i.e. that `pollForRequestedCertificate()` returns a non-empty `Optional` immediately), so it needs to be able persist the token obtained from `requestCertificate()` in a `Secret`, _and possibly complete the current reconciliation_ (i.e. issuance can be very asynchronous). 
+
+2. The issuing operator requests a new certificate. **The issuing operator puts the end entity in the `REQUESTED` state.**
+
+3. The issuing operator cannot assume that issuance will happen quickly (i.e. that `pollForRequestedCertificate()` returns a non-empty `Optional` immediately). In fact we explicitly decouple the operators reconciliation loop from the waiting for certificate issuance. If, after some number of retries within the current reconciliation, the certificate has not been issued, the token obtained from `requestCertificate()` is persisted in a `Secret` and the polling is completed in a subsequent reconciliation.
```

Comment:
I don't recall the details of theCC interaction. What I'd trying to describe in this proposal is:

1. Operator calls `issueCertificate`
2. Operator calls `pollForRequestedCertificate`, it's not ready.
3. Operator completes the rest of the reconciliation, making do without the requested certs (e.g. for renewal continuing to use the existing ones, for creation then just completing the reconciliation without having a functional cluster)
4. Operator starts a new reconciliation (by default 2 minutes later).
5. Operator calls `pollForRequestedCertificate`, it's not ready.
6. Operator completes the rest of the reconciliation making do without the requested certs
7. Operator starts a new reconciliation (by default 2 minutes later).
8. Operator calls `pollForRequestedCertificate`, this time it is ready
9. Operator does what's necessary to use the newly issued certs, maybe orchestrating trust, and then making use of the issued certificates. 

The point is that steps 4–6 could repeat many times, and in the mean time the user can update the CR and those changes get reflected, rather than cert renewal blocking changes to the CR for an arbitrarily long time.

Reply:
@ppatierno did that clarify your question?

Reply:
Maybe :-) ... the text says `after some number of retries within the current reconciliation` it seems that anyway the step 2, 5 and 8 so calling `pollForRequestedCertificate` are executed more times (retries) in the same reconcile. So we have retries in the same reconcile and if not ready, retries across reconciles.